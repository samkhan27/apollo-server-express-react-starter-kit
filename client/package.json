{
  "name": "graphql-apollo-client",
  "version": "1.0.0",
  "main": "index.js",
  "author": "Sam Khan",
  "license": "MIT",
  "scripts": {
    "start": "webpack-dev-server --open --mode development",
    "build": "webpack --mode production -p",
    "test": "jest"
  },
  "devDependencies": {
    "@babel/core": "^7.5.5",
    "@babel/plugin-proposal-class-properties": "^7.5.5",
    "@babel/plugin-transform-runtime": "^7.5.5",
    "@babel/preset-env": "^7.5.5",
    "@babel/preset-react": "^7.0.0",
    "@babel/runtime": "^7.5.5",
    "babel-jest": "^24.8.0",
    "babel-loader": "^8.0.6",
    "html-loader": "^0.5.5",
    "html-webpack-plugin": "^5.5.0",
    "jest": "^24.8.0",
    "node-fetch": "^2.6.0",
    "react-test-renderer": "^16.9.0",
    "webpack": "^4.39.2",
    "webpack-cli": "^3.3.6",
    "webpack-dev-server": "^3.8.0"
  },
  "dependencies": {
    "@apollo/react-hooks": "^3.0.0",
    "apollo-boost": "^0.4.4",
    "graphql": "^14.4.2",
    "react": "^16.9.0",
    "react-dom": "^16.9.0"
  }
}
