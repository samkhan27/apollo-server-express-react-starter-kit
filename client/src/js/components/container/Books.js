import React from 'react';
import { useQuery } from '@apollo/react-hooks';
import { gql } from 'apollo-boost';
import BookItem from '../presentational/BookItem'

const Books = () => {
  const { loading, error, data } = useQuery(gql`
    {
      books {
        title
        author
      }
    }
  `);

  if (loading) return <p>Loading...</p>;
  if (error) return <p>Error :(</p>;

  return data.books.map(({ title, author }) => (
    <BookItem key={title} title={title} author={author} />
  ));
};

export default Books;