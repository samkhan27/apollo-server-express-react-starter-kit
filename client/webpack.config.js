const HtmlWebPackPlugin = require("html-webpack-plugin")
const path = require('path')
const port = process.env.PORT || 3000

module.exports = {
  entry: './src/js/index.js',
  output: {
    filename: '[name].bundle.js',
    path: path.resolve(__dirname, 'dist')
  },
  module: {
    rules: [
      {
        test: /\.js$/,
        exclude: /node_modules/,
        use: {
          loader: "babel-loader"
        }
      },
      {
        test: /\.html$/,
        use: [
          {
            loader: "html-loader",
            options: { minimize: true }
          }
        ]
      }
    ]
  },
  plugins: [
    new HtmlWebPackPlugin({
      template: "./src/public/index.html",
      filename: "./index.html"
    })
  ],
  devtool: 'source-map',
  devServer: {
    host: 'localhost',
    port: port,
    overlay: {
      warnings: true,
      errors: true
    }
  }
}