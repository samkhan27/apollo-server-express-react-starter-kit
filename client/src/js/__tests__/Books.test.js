import React from 'react';
import Books from '../components/container/Books';
import { create, act } from 'react-test-renderer';
import { ApolloProvider } from '@apollo/react-hooks';
import client from '../services/graphql/apolloClient';

test('It loads the book component', () => {
  let component;
  act(() => {
    component = create(
      <ApolloProvider client={client}>
        <Books />
      </ApolloProvider>
    );
  });

  expect(component.toJSON()).toMatchSnapshot();
});
