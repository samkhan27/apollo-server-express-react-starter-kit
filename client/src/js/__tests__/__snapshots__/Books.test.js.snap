// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It loads the book component 1`] = `
<p>
  Loading...
</p>
`;
